File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> numberVsFrequencyMap = new HashMap<>();
        for (int num: nums) {
            numberVsFrequencyMap.put(num, numberVsFrequencyMap.getOrDefault(num, 0) + 1);
        }
        List<Map.Entry<Integer, Integer>> keyValueEntryMapList = new ArrayList<>(numberVsFrequencyMap.entrySet());
        keyValueEntryMapList.sort((a, b) -> {
            return Integer.compare(b.getValue(), a.getValue());
        });
        int[] result = new int[k];
        for (int i = 0; i < k; i++) {
            result[i] = keyValueEntryMapList.get(i).getKey();
        }
        return result;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        dict = {}
        for num in nums:
            if num in dict:
                dict[num] += 1
            else:
                dict[num] = 1
        for value in dict.values():
            if value > 1:
                return True
        return False